### DISCUSSION TITLE
No combinations in Friesach

### DISCUSSION THREAD

**Discussion by jrwood54** [GrandOrgue/grandorgue#1642] [Help]
I'm new to GrandOrgue and have been trying out the Friesach sample set.  It opens without any active pistons and I can't figure out how to program combinations.  Any help would be greatly appreciated.

**Comment by ClarkeBrunt** [comment]
Hi. I'm going to assume you know how pistons normally work on actual organs, and that you've got some kind of MIDI console.

The Friesach sample set appears not to have any pistons on the supplied panels. There is a row of 'buttons' under the bottom manual, and a few things above the pedal board, but they seem to be just for decoration - unless I missed something. No sign that they are programmable for MIDI by right clicking on them.

So you need to use GO's supplied panels - Panel | Divisionals, Panel | Generals, etc. It might be useful also to read the help about these panels.

Just using the on-screen interface for now: Choose the stops you're trying to program onto a piston; put GO into 'Set' mode ('Set' button on those panels, or far left of toobar); press the piston you're programming; take out of 'Set' mode, and you're done (unless going on to program other ones).

If you want actual buttons on your console to do this stuff, then right-click on the piston, set button, or whatever; hit 'Detect complex MIDI event'; and follow the on-screen instructions. You can then use your physical buttons to do the programming etc.

It is usual (I believe) on an actual console that the 'Set' button enters set mode when pressed, and leaves set mode when released, so you actually have to hold it down, and can't forget to exit set mode. So as well to program your button like that - so when the 'Detect MIDI' says "Please toggle it (to on state if possible)" then press and hold until it says "Please toggle it again (to off state, if possible", then release it.

Incidentally, when I first used the Friesach organ, I made some observations here #1407 . The sample set doesn't have a 'General Cancel' (GC) button, but there is one is some of the GO panels which you can program - but it turns off some of the action noises and then there is no (obvious) way to turn them back on. Maybe counts as a bug in the sample set - I fixed it, and some other things, by editing the ODF file. As I mentioned in that other thread, the sample set does have an on-screen blower switch - far right thing above the top manual - it's so small that it's difficult to select it - but it does turn the blower off/on (lots of bellows sounds, and no sound when off!).

**Comment by eturpault** [comment]
I share an ODF of Friesach in which I have added a panel called "Single jamb" having general and sequencer pistons (which can be controlled by MIDI). It uses built-in bitmaps of GrandOrgue, but the background which uses the info image of the sample set and which can be replaced by a custom image in the section Panel002Image001.

[Friesach single jamb ODF.zip](https://github.com/GrandOrgue/grandorgue/files/12446736/Friesach.single.jamb.ODF.zip)

![Friesach single jamb panel](https://github.com/GrandOrgue/grandorgue/assets/51839171/3f586306-c5cc-4b2f-9cf8-922f4ee89f44)

**Comment by organistaj72** [comment]
Witam. Czy możesz przedstawić ogólnie w jaki sposób tworzyć taki panel? Od czego zacząć?

**Reply by eturpault** [reply]
For a simple way to generate panels you could use the GOODF tool of @larspalo, and study how such a panel is made to replicate the elements you want to have in your own panel.

**Reply by organistaj72** [reply]
Dzięki za odpowiedź. Przetestuję i odezwę się.

**Reply by organistaj72** [reply]
Witam.  Czy jest możliwość kontaktu telefonicznego z Tobą czy tylko przez stronę.

**Reply by eturpault** [reply]
Hello, only by this website, and in English please (I am French and using Google translation to understand your messages).

**Reply by organistaj72** [reply]
Dzięki. Pozdrawiam.

**Comment by jason-barba** [comment]
Here is my contribution. I have added a selection of combination buttons to the main panel [Organ].

//Jens

![Screenshot from 2024-02-06 08-41-58](https://github.com/GrandOrgue/grandorgue/assets/56149267/abeee7ff-7202-4c5f-a579-27ffc1aa3646)

Extracts from the .organ file:
.
.

[Organ]
.
.
;*********************
; NumberOfSetterElements=0
NumberOfSetterElements=9
;*********************
.
.
;**************************************
[SetterElement001]
Type=Label
PositionX=535
PositionY=140
DispLabelFontSize=16
DispLabelColour=BLACK

[SetterElement002]
Type=Save
PositionX=1100
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
DispLabelText=SAVE
DispLabelFontSize=6
DispLabelColour=BLACK

[SetterElement003]
Type=P10
PositionX=640
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
DispLabelFontSize=8
DispLabelColour=BLACK

[SetterElement004]
Type=M1
PositionX=520
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
;DispLabelText=-1
DispLabelFontSize=8
DispLabelColour=BLACK

[SetterElement005]
Type=P1
PositionX=600
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
;DispLabelText=+1
DispLabelFontSize=8
DispLabelColour=BLACK

[SetterElement006]
Type=M10
PositionX=480
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
DispLabelFontSize=8
DispLabelColour=BLACK

[SetterElement007]
Type=Set
PositionX=1100
PositionY=60
DisplayAsPiston=Y
DispImageNum=2
DispLabelText=SET
DispLabelFontSize=6
DispLabelColour=BLACK

[SetterElement008]
Type=Current
PositionX=560
PositionY=10
DisplayAsPiston=Y
DispImageNum=2
DispLabelText=CUR
DispLabelFontSize=6
DispLabelColour=BLACK

[SetterElement009]
Type=GC
PositionX=1100
PositionY=140
DisplayAsPiston=Y
DispImageNum=2
DispLabelText=GC
DispLabelFontSize=6
DispLabelColour=BLACK
;*****************************************
